http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoMetricsMonitor.scala

```diff
@@ -0,0 +1,342 @@
+package org.broadinstitute.dsde.workbench.leonardo
+package monitor
+
+import cats.effect.Async
+import cats.effect.implicits.concurrentParTraverseOps
+import cats.mtl.Ask
+import cats.syntax.all._
+import fs2.Stream
+import org.broadinstitute.dsde.workbench.azure.AzureCloudContext
+import org.broadinstitute.dsde.workbench.google2.KubernetesSerializableName.ServiceName
+import org.broadinstitute.dsde.workbench.leonardo.config.Config
+import org.broadinstitute.dsde.workbench.leonardo.dao.{ToolDAO, _}
+import org.broadinstitute.dsde.workbench.leonardo.db.{clusterQuery, DbReference, KubernetesServiceDbQueries}
+import org.broadinstitute.dsde.workbench.leonardo.http.{dbioToIO, _}
+import org.broadinstitute.dsde.workbench.leonardo.monitor.LeoMetric._
+import org.broadinstitute.dsde.workbench.leonardo.util.AppCreationException
+import org.broadinstitute.dsde.workbench.model.TraceId
+import org.broadinstitute.dsde.workbench.openTelemetry.OpenTelemetryMetrics
+import org.http4s.headers.Authorization
+import org.http4s.{AuthScheme, Credentials, Uri}
+import org.typelevel.log4cats.StructuredLogger
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.FiniteDuration
+
+/** Collects metrics about active Leo runtimes and apps. */
+class LeoMetricsMonitor[F[_]](config: LeoMetricsMonitorConfig,
+                              appDAO: AppDAO[F],
+                              wdsDAO: WdsDAO[F],
+                              cbasDAO: CbasDAO[F],
+                              cbasUiDAO: CbasUiDAO[F],
+                              cromwellDAO: CromwellDAO[F],
+                              samDAO: SamDAO[F]
+)(implicit
+  F: Async[F],
+  dbRef: DbReference[F],
+  metrics: OpenTelemetryMetrics[F],
+  logger: StructuredLogger[F],
+  clusterToolToToolDao: RuntimeContainerServiceType => ToolDAO[F, RuntimeContainerServiceType],
+  ec: ExecutionContext
+) {
+  private val parallelism = 25
```

Comment:
Where does the 25 come from?

Reply:
A guess! Seemed like a reasonable parallelism limit. :-)